lib/subsetFonts.js

```diff
@@ -728,26 +728,29 @@ async function subsetFonts(
   }
 
   if (!subsetPerPage) {
-    const globalFontUsage = {};
-
-    // Gather all texts
-    for (const htmlAssetTextWithProps of htmlAssetTextsWithProps) {
-      for (const fontUsage of htmlAssetTextWithProps.fontUsages) {
-        if (!globalFontUsage[fontUsage.fontUrl]) {
-          globalFontUsage[fontUsage.fontUrl] = [];
+    const fontUsagesConcat = _.flatMap(
+      htmlAssetTextsWithProps,
+      htmlAssetTextWithProps => htmlAssetTextWithProps.fontUsages
+    );
+    const globalFontUsageMap = fontUsagesConcat.reduce(
```

Comment:
Mind turning this `reduce` construct into a regular `for`...`of` loop? I find those much easier to reason about.